internal/executor/contracts/interchain.go

```diff
@@ -66,206 +43,33 @@ func (chain *appchain) UnmarshalJSON(data []byte) error {
 		t.SourceReceiptCounter = make(map[string]uint64)
 	}
 
-	*chain = appchain(*t)
+	*i = Interchain(*t)
 	return nil
 }
 
-// Register appchain manager registers appchain info caller is the appchain
-// manager address return appchain id and error
-func (x *Interchain) Register(validators string, consensusType int32, chainType, name, desc, version, pubkey string) *boltvm.Response {
-	chain := &appchain{
-		ID:            x.Caller(),
-		Name:          name,
-		Validators:    validators,
-		ConsensusType: consensusType,
-		ChainType:     chainType,
-		Desc:          desc,
-		Version:       version,
-		PublicKey:     pubkey,
-	}
-
+func (x *InterchainManager) Register() *boltvm.Response {
+	interchain := &Interchain{ID: x.Caller()}
 	ok := x.Has(x.appchainKey(x.Caller()))
 	if ok {
-		x.Stub.Logger().WithFields(logrus.Fields{
-			"id": x.Caller(),
-		}).Debug("Appchain has registered")
-		x.GetObject(x.appchainKey(x.Caller()), chain)
+		x.GetObject(x.appchainKey(x.Caller()), interchain)
 	} else {
-		// logger.Info(x.Caller())
-		x.SetObject(x.appchainKey(x.Caller()), chain)
-		x.Logger().WithFields(logrus.Fields{
-			"id": x.Caller(),
-		}).Info("Appchain register successfully")
+		x.SetObject(x.appchainKey(x.Caller()), interchain)
 	}
-	body, err := json.Marshal(chain)
+	body, err := json.Marshal(interchain)
 	if err != nil {
 		return boltvm.Error(err.Error())
 	}
 
 	return boltvm.Success(body)
 }
 
-func (x *Interchain) UpdateAppchain(validators string, consensusType int32, chainType, name, desc, version, pubkey string) *boltvm.Response {
-	ok := x.Has(x.appchainKey(x.Caller()))
-	if !ok {
-		return boltvm.Error("register appchain firstly")
-	}
-
-	chain := &appchain{}
-	x.GetObject(x.appchainKey(x.Caller()), chain)
-
-	if chain.Status == registered {
-		return boltvm.Error("this appchain is being audited")
-	}
-
-	chain = &appchain{
-		ID:            x.Caller(),
-		Name:          name,
-		Validators:    validators,
-		ConsensusType: consensusType,
-		ChainType:     chainType,
-		Desc:          desc,
-		Version:       version,
-		PublicKey:     pubkey,
-	}
-
-	x.SetObject(x.appchainKey(x.Caller()), chain)
-
+func (x *InterchainManager) DeleteInterchain(id string) *boltvm.Response {
+	x.Delete(appchainMgr.PREFIX + id)
```

Comment:
suggest to use x.appchainKey(id) instead of appchainMgr.PREFIX + id

Reply:
fixed